### PR TITLE
Fix Windows ARM64 build

### PR DESCRIPTION
**Pull request by AntonLapounov**

### CODE REVIEW

File: src/coreclr/nativeaot/Runtime/windows/CoffNativeCodeManager.cpp [hunk 1]
```diff
@@ -554,11 +559,41 @@ bool CoffNativeCodeManager::UnwindStackFrame(MethodInfo *    pMethodInfo,
     pRegisterSet->pIP = PTR_PCODE(pRegisterSet->SP - sizeof(TADDR));
 
     memcpy(pRegisterSet->Xmm, &context.Xmm6, sizeof(pRegisterSet->Xmm));
-#endif // TARGET_X86
+#elif defined(TARGET_ARM64)
+    for (int i = 8; i < 16; i++)
+    {
+        context.V[i].Low = pRegisterSet->D[i - 8];
+        context.V[i].High = 0;
```
Comment (AntonLapounov) [inline src/coreclr/nativeaot/Runtime/windows/CoffNativeCodeManager.cpp:566]: Do we need to copy NEON registers here? We do copy XMM6-XMM15 for x64.

File: src/coreclr/nativeaot/Runtime/windows/CoffNativeCodeManager.cpp [hunk 2]
```diff
@@ -554,11 +559,41 @@ bool CoffNativeCodeManager::UnwindStackFrame(MethodInfo *    pMethodInfo,
...
+                    (PRUNTIME_FUNCTION)pNativeMethodInfo->runtimeFunction,
+                    &context,
+                    &HandlerData,
+                    &EstablisherFrame,
+                    &contextPointers);
+
+    pRegisterSet->SP = context.Sp;
+    pRegisterSet->IP = context.Pc;
+
+    pRegisterSet->pIP = PTR_PCODE(pRegisterSet->SP - sizeof(TADDR));
```
Comment (jkotas) [inline src/coreclr/nativeaot/Runtime/windows/CoffNativeCodeManager.cpp:587]: How do we know that `pIP` is `SP - sizeof(TADDR)`? I do not think that this is a valid assumption on ARM64.

File: src/coreclr/nativeaot/Runtime/windows/CoffNativeCodeManager.cpp (hunk 1 above)
Comment (jkotas) [reply src/coreclr/nativeaot/Runtime/windows/CoffNativeCodeManager.cpp:566]: We only need to track non-volatile registers (https://docs.microsoft.com/en-us/cpp/build/arm64-windows-abi-conventions?view=msvc-160) here.

File: src/coreclr/nativeaot/Runtime/windows/CoffNativeCodeManager.cpp [hunk 3]
```diff
@@ -554,11 +559,41 @@ bool CoffNativeCodeManager::UnwindStackFrame(MethodInfo *    pMethodInfo,
     pRegisterSet->pIP = PTR_PCODE(pRegisterSet->SP - sizeof(TADDR));
 
     memcpy(pRegisterSet->Xmm, &context.Xmm6, sizeof(pRegisterSet->Xmm));
-#endif // TARGET_X86
+#elif defined(TARGET_ARM64)
+    for (int i = 8; i < 16; i++)
+    {
+        context.V[i].Low = pRegisterSet->D[i - 8];
+        context.V[i].High = 0;
+    }
```
Comment (jkotas) [inline src/coreclr/nativeaot/Runtime/windows/CoffNativeCodeManager.cpp:567]:
```suggestion
    for (int i = 8; i < 16; i++)
        context.V[i].Low = pRegisterSet->D[i - 8];
```

**Comment by jkotas** [comment]
LGTM otherwise.

File: src/coreclr/nativeaot/Runtime/windows/CoffNativeCodeManager.cpp (hunk 2 above)
Comment (AntonLapounov) [reply src/coreclr/nativeaot/Runtime/windows/CoffNativeCodeManager.cpp:587]: Good catch, I guess it should be `pRegisterSet->pIP = pRegisterSet->pLR`.

File: src/coreclr/nativeaot/Runtime/windows/CoffNativeCodeManager.cpp (hunk 1 above)
Comment (AntonLapounov) [reply src/coreclr/nativeaot/Runtime/windows/CoffNativeCodeManager.cpp:566]: I meant that if we use this function for GC stack walking only and we never save addresses to `D` registers, we could probably skip even non-volatile `D8`-`D15` ones.

Comment (jkotas) [reply src/coreclr/nativeaot/Runtime/windows/CoffNativeCodeManager.cpp]: This is also used for exception handling where we need to restore the non-volatile registers.

**Review by jkotas — APPROVED** [review]
Thanks!

**Pull request merged**